e2e/test/scenarios/sharing/public-question.cy.spec.js

```diff
@@ -140,6 +136,64 @@ describe("scenarios > public > question", () => {
       });
     }),
   );
+
+  it("should be able to view public questions with snippets", () => {
+    cy.visit("/");
+
+    openNativeEditor();
+
+    // Create a snippet
+    cy.icon("snippet").click();
+    cy.findByTestId("sidebar-content").findByText("Create a snippet").click();
+
+    modal().within(() => {
+      cy.findByLabelText("Enter some SQL here so you can reuse it later").type(
+        `'test'`,
+      );
+      cy.findByLabelText("Give your snippet a name").type("string 'test'");
+      cy.findByText("Save").click();
+    });
+
+    cy.get("@editor").type(`{moveToStart}select `);
+
+    saveQuestion("test question", { wrapId: true });
+    cy.get("@questionId").then(id => {
+      cy.request("POST", `/api/card/${id}/public_link`).then(
+        ({ body: { uuid } }) => {
+          cy.visit(`/public/question/${uuid}`);
+          cy.get(".cellData").contains("test");
+        },
+      );
+    });
+  });
+
+  it("should be able to view public questions with card template tags", () => {
+    cy.createNativeQuestion({
+      name: "Nested Question",
+      native: {
+        query: "SELECT * FROM PEOPLE",
```

Comment:
Our e2e tests will run a bit faster (which pays out in the long run) if we limit the amount of data exchanged in API requests:

```suggestion
        query: "SELECT * FROM PEOPLE LIMIT 5",
```